cypress/e2e/keymaps.cy.js

```diff
@@ -314,4 +349,1450 @@ describe('Keymaps', () => {
+              // getState("activeStep") > 1 &&
+              setState("activeStep", getState("activeStep") - 1)
+            }
+          >
+            Text
+          </button>
+        `);
+      });
+    });
+
+    describe('should wrap a single line selection in a comment', () => {
+      describe('standard', () => {
+        it('block', () => {
+          loadPlayroom(blockStarter);
+          selectToEndOfLine();
+
+          typeComment();
+
+          assertCodePaneContains(dedent`
+            {/* <div>First line</div> */}
+            <div>Second line</div>
+            <div>Third line</div>
+          `);
+        });
+
+        it('line', () => {
+          loadPlayroom(lineStarter);
+
+          moveBy(0, 1);
+          selectToEndOfLine();
+
+          typeComment();
+
+          assertCodePaneContains(dedent`
+            <div
+              // prop1="This is the first prop"
+              prop2="This is the second prop"
+              prop3="This is the third prop"
+            >
+              First line
+            </div>
+            <div>Second line</div>
+            <div>Third line</div>
+          `);
+        });
+      });
+
+      describe('without shifting selection position for a forward selection', () => {
+        it('block', () => {
+          loadPlayroom(blockStarter);
+          selectToEndOfLine();
+
+          typeComment();
+
+          typeCode(`{shift+leftArrow}`);
+          typeCode('c');
+
+          assertCodePaneContains(dedent`
+            {/* c> */}
+            <div>Second line</div>
+            <div>Third line</div>
+          `);
+        });
+
+        it('line', () => {
+          loadPlayroom(lineStarter);
+
+          moveBy(0, 1);
+          moveByWords(1);
+          selectToEndOfLine();
+
+          typeComment();
+
+          typeCode(`{shift+leftArrow}`);
+          typeCode('c');
+
+          assertCodePaneContains(dedent`
+            <div
+              // c"
+              prop2="This is the second prop"
+              prop3="This is the third prop"
+            >
+              First line
+            </div>
+            <div>Second line</div>
+            <div>Third line</div>
+          `);
+        });
+      });
+
+      describe('without shifting selection position for a backward selection', () => {
+        it('block', () => {
+          loadPlayroom(blockStarter);
+
+          moveToEndOfLine();
+          selectToStartOfLine();
+
+          typeComment();
+
+          typeCode(`{shift+rightArrow}`);
+          typeCode('c');
+
+          assertCodePaneContains(dedent`
+            {/* <c */}
+            <div>Second line</div>
+            <div>Third line</div>
+          `);
+        });
+
+        it('line', () => {
+          loadPlayroom(lineStarter);
+
+          moveBy(0, 1);
+          moveToEndOfLine();
+
+          // Todo - (1/2) Solve issue where Ubuntu does not select to the start of line
+          // Todo - (2/2) with one trigger of the keybinding
```

Comment:
Has this been addressed? Is this an ubuntu issue, or a broader linux issue?

Reply:
This hasn't been addressed. I'm not 100% sure who the culprit is but by default on the Ubuntu tests, the backwards selection goes up to the opening "//" (not including), whereas macOS selection goes to the very start of the line.